pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,408 @@
+package org.broadinstitute.dsm.db;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.structure.ColumnName;
+import org.broadinstitute.dsm.db.structure.TableName;
+import org.broadinstitute.dsm.model.elastic.export.Exportable;
+import org.broadinstitute.dsm.model.elastic.export.painless.PutToNestedScriptBuilder;
+import org.broadinstitute.dsm.model.elastic.export.painless.UpsertPainlessFacade;
+import org.broadinstitute.dsm.model.filter.postfilter.HasDdpInstanceId;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+@Data
+@TableName(name = DBConstants.SOMATIC_DOCUMENTS_TABLE,
+        alias = DBConstants.SOMATIC_DOCUMENTS_TABLE_ALIAS,
+        primaryKey = DBConstants.SOMATIC_DOCUMENTS_PK,
+        columnPrefix = "")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SomaticResultUpload implements HasDdpInstanceId {
+    @ColumnName(ESObjectConstants.SOMATIC_DOCUMENT_ID)
+    private final Long somaticDocumentId;
+
+    @ColumnName(DBConstants.DDP_INSTANCE_ID)
+    private final Long ddpInstanceId;
+
+    private final String ddpParticipantId;
+
+    @ColumnName(DBConstants.PARTICIPANT_ID)
```

Comment:
Why do we need both ddpParticipantId and participantId?

Reply:
We need the internal ID when doing DSM related key constraints and the ddpParticipantId to communicate with DSS for PEPPER-903.

Reply:
I'm storing the `DSM` participant ID and joining to get the DDP Participant ID.  See `SQL_SELECT_DOCUMENTS_BASE`

Reply:
Maybe we can discuss offline? I'm still not sure why we need to store DDP participant ID here.
And it seems like we will be deep into the code before we discover a bad DDP participant ID was used in the request.

Reply:
Thx for the DM explanation. This was my misunderstanding of the columnname annotation.
We are NOT storing the DDP participant ID in SOMATIC_DOCUMENTS_TABLE, which was my concern.